src/python/pants/backend/python/util_rules/pex_environment.py

```diff
@@ -124,49 +129,26 @@ def __post_init__(self) -> None:
     def message(self) -> str:
         return f"Selected {self.path} to run PEXes with."
 
-    @classmethod
-    def from_python_binary(cls, python_binary: PythonBinary) -> PythonExecutable:
-        """Converts from PythonBinary to PythonExecutable.
-
-        The PythonBinary type is a singleton representing the Python that is used for script
-        execution by `@rule`s. On the other hand, there may be multiple PythonExecutables, since
-        they are subject to a user's interpreter constraints.
-        """
-        return cls(path=python_binary.path, fingerprint=python_binary.fingerprint)
-
 
 @dataclass(frozen=True)
 class PexEnvironment(EngineAwareReturnType):
```

Comment:
there's nothing left from what I can see in this class that requires the `EngineAwareReturnType`.